File: src/main/java/Monotonic.java
import java.util.HashSet;
import java.util.Set;

public class Monotonic {
    public boolean isMonotonic(int[] nums) {
        boolean increasing = true;
        boolean decreasing = true;

        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                increasing = false;
            } else if (nums[i] < nums[i + 1]) {
                decreasing = false;
            }
        }
        return !(increasing == true && decreasing == true);
    }
}

File: src/main/kotlin/com/corrigan/Monotonic.kt
package com.corrigan

object Monotonic {
    fun isMonotonic(ints: IntArray): Boolean {
        val state = mutableSetOf<String>()
        ints.toList().windowed(size = 2, step = 1).forEach {
            when {
                it[0] > it[1] -> state.add("decreasing")
                it[0] < it[1] -> state.add("increasing")
                else -> state.add("same")
            }
        }
        return state.containsAll(setOf("increasing", "decreasing")).not()
    }

}